graphics/src/SubMesh.cc

```diff
@@ -220,7 +243,17 @@ bool SubMesh::HasNormal(const unsigned int _index) const
 //////////////////////////////////////////////////
 bool SubMesh::HasTexCoord(const unsigned int _index) const
 {
-  return _index < this->dataPtr->texCoords.size();
+  if (this->dataPtr->texCoords.empty())
+    return false;
+  unsigned firstSetIndex = this->dataPtr->texCoords.begin()->first;
+  return this->HasTexCoordBySet(_index, firstSetIndex);
+}
+
+//////////////////////////////////////////////////
+bool SubMesh::HasTexCoordBySet(unsigned int _index,
+    unsigned int _setIndex) const
+{
+  return _index < this->dataPtr->texCoords[_setIndex].size();
```

Comment:
Should we check if `texCoords` has `setIndex` first?

There are more instances below.

Reply:
yep, added checks. 50c0e2e